### ISSUE TITLE
Subsonic: do not return the coverArt id if it doesn't exist

### ISSUE THREAD

**Issue by cquike** [ampache/ampache#3149]
### Describe the bug

When using the subsonic API a request like this /rest/getAlbum.view?id=200000XXX will return a JSON object which includes the coverArt field _even_ if that album doesn't have a cover. That leads to potential errors in the client when they try to access the cover with getCoverArt. Since the coverArt field in the subsonic API is optional I think it would be better not to fill it when there is no cover. 

## Environment

* Ampache version: 5.1.0
* Web server + version: apache
* Server operating system: Raspberry os 11
* Client operating system: Debian 11

### Client type

* SubSonic (strawberry player)

**Comment by cquike** [comment]
I have created a PR #3159 that would partially fix the problem.

However the response from getAlbum is still not fully correct: The song items within the album _still_ contain the covertArt attribute. However I have avoided touching the createSong() function under Subsonic_Xml_Data.php because it looks rather different to the rest of the functions in the same file like addAlbm(), addPlaylist(), addArtist(). In these other functions an Album, Playlist or Artist object is created and then the information is filled from the attributes of that object. However addSong() retrieves information directly from the database, which seems to me like the wrong thing to do, because in the rest of ampache all the database access is restricted to the Repository module. Is there maybe a reason (performance?) for such discrepancy?

Also, I have not modified the rest of the coverArt for getArtist, getPlaylist, ... which I think are also affected by the same problem.
